Linux aarch64 support. Update to 1.9.7

martin-g opened the pull request:
This PR is based on top of https://github.com/conda-forge/pv-feedstock/pull/9.
It applies the linter suggestions from https://github.com/conda-forge/pv-feedstock/pull/9#issuecomment-2265012564, updates PV to 1.8.12 to fix the OSX build, and skips one failing test on OSX.

Checklist
* [x] Used a fork of the feedstock to propose changes
* [x] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/conda_smithy.html#how-to-re-render ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

<!--
Please note any issues this fixes using [closing keywords]( https://help.github.com/articles/closing-issues-using-keywords/ ):
-->

<!--
Please add any other relevant info below:
-->

regro-cf-autotick-bot committed:
migration: ArchRebuild

regro-cf-autotick-bot committed:
MNT: Re-rendered with conda-build 24.7.1, conda-smithy 3.37.2, and conda-forge-pinning 2024.08.02.08.34.06

martin-g committed:
Fix linting issues

Applies suggestions from https://github.com/conda-forge/pv-feedstock/pull/9#issuecomment-2265012564

Signed-off-by: Martin Tzvetanov Grigorov <mgrigorov@apache.org>

martin-g commented:
@conda-forge-admin, please rerender

Commit:
MNT: Re-rendered with conda-build 24.7.1, conda-smithy 3.37.2, and conda-forge-pinning 2024.08.04.09.56.37

martin-g committed:
Update PV to 1.8.2

Hoping that this will fix the OSX build

Signed-off-by: Martin Tzvetanov Grigorov <mgrigorov@apache.org>

martin-g committed:
Fix the file extension

Signed-off-by: Martin Tzvetanov Grigorov <mgrigorov@apache.org>

martin-g committed:
Try to build PPC64LE at Azure (cross-compile)

Signed-off-by: Martin Tzvetanov Grigorov <mgrigorov@apache.org>

martin-g commented:
@conda-forge-admin, please rerender

Commit:
MNT: Re-rendered with conda-build 24.7.1, conda-smithy 3.37.2, and conda-forge-pinning 2024.08.04.09.56.37

martin-g committed:
Try to build osx-64 on linux-64 (cross-compilation)

A test is failing on OSX 64:
```
FAIL: tests/Display_-_--numeric_--bytes.test
```

Signed-off-by: Martin Tzvetanov Grigorov <mgrigorov@apache.org>

martin-g commented:
@conda-forge-admin, please rerender

Commit:
MNT: Re-rendered with conda-build 24.7.1, conda-smithy 3.37.2, and conda-forge-pinning 2024.08.05.11.36.16

martin-g committed:
Use a patch to skip one test on OSX

Signed-off-by: Martin Tzvetanov Grigorov <mgrigorov@apache.org>

martin-g commented:
@conda-forge-admin, please rerender

Commit:
MNT: Re-rendered with conda-build 24.7.1, conda-smithy 3.37.2, and conda-forge-pinning 2024.08.05.11.36.16

martin-g commented:
All green!

@conda-forge-admin, please ping team

wolfv commented:
This looks good to me. Do you want to add yourself as a maintainer?

wolfv commented on recipe/meta.yaml:
```diff
@@ -1,24 +1,27 @@
 source:
-  fn: {{ pkgname }}-{{ version }}.tar.bz2
-  url: http://www.ivarch.com/programs/sources/{{ pkgname }}-{{ version }}.tar.bz2
+  fn: {{ pkgname }}-{{ version }}.tar.gz
```
this can be removed.

martin-g replied:
Done!

wolfv commented on recipe/meta.yaml:
```diff
@@ -1,24 +1,27 @@
-  url: http://www.ivarch.com/programs/sources/{{ pkgname }}-{{ version }}.tar.bz2
+  fn: {{ pkgname }}-{{ version }}.tar.gz
+  url: http://www.ivarch.com/programs/sources/{{ pkgname }}-{{ version }}.tar.gz
   sha256: {{ sha256 }}
```
sha can go directly in here.

martin-g replied:
Done!

martin-g committed:
Apply code review suggestions

- add myself as a co-maintainer
- inline sha256
- remove `source > fn`

martin-g commented:
> Do you want to add yourself as a maintainer?

Added myself!

martin-g commented:
@conda-forge-admin, please rerender

Commit:
MNT: Re-rendered with conda-build 24.11.2, conda-smithy 3.44.9, and conda-forge-pinning 2024.12.05.23.59.20

martin-g committed:
Update pv to 1.9.7 (latest stable)

Hopefully the test suite now will pass on OSX x86_64 too

Signed-off-by: Martin Tzvetanov Grigorov <mgrigorov@apache.org>

martin-g committed:
Temporarily disable the patch that skips tests

Signed-off-by: Martin Tzvetanov Grigorov <mgrigorov@apache.org>

martin-g committed:
Do not build for linux-ppc64le.

It fails with:
```
checking for uid_t in sys/types.h... yes
checking for struct stat.st_rdev... yes
configure: error: in `/home/conda/feedstock_root/build_artifacts/pv_1733472558323/work':
configure: error: cannot run test program while cross compiling
See `config.log' for more details
Traceback (most recent call last):
  File "/opt/conda/lib/python3.12/site-packages/conda_build/build.py", line 2557, in build
    utils.check_call_env(
  File "/opt/conda/lib/python3.12/site-packages/conda_build/utils.py", line 406, in check_call_env
    return _func_defaulting_env_to_os_environ("call", *popenargs, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/conda/lib/python3.12/site-packages/conda_build/utils.py", line 382, in _func_defaulting_env_to_os_environ
    raise subprocess.CalledProcessError(proc.returncode, _args)
```

Signed-off-by: Martin Tzvetanov Grigorov <mgrigorov@apache.org>

martin-g committed:
Re-enable the patch for OSX to not run a specific test

Signed-off-by: Martin Tzvetanov Grigorov <mgrigorov@apache.org>

martin-g commented:
@conda-forge-admin, please rerender

martin-g commented:
@conda-forge-admin, please rerender

Commit:
MNT: Re-rendered with conda-build 24.11.2, conda-smithy 3.44.9, and conda-forge-pinning 2024.12.06.08.06.51

martin-g committed:
Add one more test to the skipped ones for osx

Signed-off-by: Martin Tzvetanov Grigorov <mgrigorov@apache.org>

martin-g commented:
@wolfv The PR is ready for another review !
